### ISSUE TITLE
bevy memory usage grows over time

### ISSUE THREAD

**Issue by Byteron** [bevyengine/bevy#5691]
## Bevy version

0.8

## What went wrong

I left a bevy app running in the background for a long time, and well, it grew to 12 GB memory usage over the course of ~8 hours.
The App is just a few planes and a simple controller system.

![image](https://user-images.githubusercontent.com/24433899/184551225-f35c3b4f-381d-4e75-9def-1b418b4a1a32.png)

**Comment by alice-i-cecile** [comment]
This has been examined before; IIRC this is the result of logging enabled during debug builds.

**Comment by Byteron** [comment]
I can confirm that this is only happening in debug builds. when `cargo run --release` the mem usage is not growing.

**Comment by afonsolage** [comment]
I let a game with only Default plugins running for 1 hour, with `cargo run --release` on Windows 11 and it is still increasing memory usage.

```rust
use bevy::prelude::*;

fn main() {
    App::new().add_plugins(DefaultPlugins).run();
}
```

![image](https://user-images.githubusercontent.com/1176452/184596970-15fb32e7-12d8-4fce-b5a2-d52e1f3a2093.png)
_On startup_

![image](https://user-images.githubusercontent.com/1176452/184597032-196895a8-434b-4039-b038-a7ea3fb65e3e.png)
_1 hour later_

| Running Time| Memory Usage |
| --- | --- |
|0h | 65.9 MB |
|1h | 150.6 MB |

**Comment by alice-i-cecile** [comment]
Moving this to a docs issue then :)

**Comment by bjorn3** [comment]
> It seems this is some internal memory/cache pool. It noticed when it was about to reach 500 MB it dropped to ~50 MB again and restarted the cycle.

Does it also drop down in case of memory pressure from other program? If so, that should be fine, if confusing for end users. If not, it is probably still an issue.

**Comment by afonsolage** [comment]
> > It seems this is some internal memory/cache pool. It noticed when it was about to reach 500 MB it dropped to ~50 MB again and restarted the cycle.
> 
> Does it also drop down in case of memory pressure from other program? If so, that should be fine, if confusing for end users. If not, it is probably still an issue.

![image](https://user-images.githubusercontent.com/1176452/184664132-fb9d65a3-478c-4481-81bd-9cd86e9f26ec.png)

It was using around 800MB and then I launched 15 [wasm bevymark](https://bevyengine.org/examples/stress-tests/bevymark/) with 500k sprites each to force Chrome do what it does better: eating ram and than the `reflect_example` which is the program that is running for about 12hs dropped it's memory to about 100MB.

After a while it crashed, since the Windows ran out of ram:
```
memory allocation of 9591008 bytes failed
error: process didn't exit successfully: `target\release\reflect_example.exe` (exit code: 0xc0000409, STATUS_STACK_BUFFER_OVERRUN)
```

**Comment by hammypants** [comment]
> This has been examined before; IIRC this is the result of logging enabled during debug builds.

Is there a way to disable this?

```rust
fn main() {
    App::default().add_plugins_with(DefaultPlugins, |group| { group.disable::<bevy::log::LogPlugin>() }).run();
}
```

Still grows at an exceptional rate with `cargo run`.

**Comment by alice-i-cecile** [comment]
Yep, we're getting another memory leak over in https://github.com/bevyengine/bevy/issues/6417: the logging is a factor, but far from the only issue.

**Comment by james7132** [comment]
#6878 seems to fix the issue seen in #6417. @Byteron can you check if it fixes this leak?

**Comment by Yairama** [comment]
Maybe this will help someone, that's why I'm commenting.

When I installed Vulkan, I started to get a Validation layers error, an error that does not stop the execution but causes a slight increase in memory over time.
 The error was:
 
`ERROR wgpu_hal::vulkan::instance: VALIDATION [UNASSIGNED-CoreValidation-Shader-InconsistentSpirv (0x6bbb14)]
	Validation Error: [ UNASSIGNED-CoreValidation-Shader-InconsistentSpirv ] Object 0: handle = 0x24ebcd349f0, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0x6bbb14 | SPIR-V module not valid: [VUID-StandaloneSpirv-OpTypeRuntimeArray-04680] Vulkan, OpTypeStruct containing an OpTypeRuntimeArray must be decorated with Block or BufferBlock.
  %PointLights = OpTypeStruct %_runtimearr_PointLight`

It was solved by removing vulkan completely.

**Comment by TotalKrill** [comment]
On webbrowsers this results in crashed tabs after the app has been running for a while. However, we seem to be able to work around this by using: 

``` rust
DefaultPlugins.build()
         .disable::<LogPlugin>()
         .disable::<DiagnosticsPlugin>();

```
